connectors/sources/jira.py

```diff
@@ -320,8 +371,310 @@ def get_default_configuration(cls):
+
+        Returns:
+            bool: True if document level security is enabled, False otherwise.
+        """
+        if self._features is None:
+            return False
+
+        if not self._features.document_level_security_enabled():
+            return False
+
+        return self.configuration["use_document_level_security"]
+
+    def access_control_query(self, access_control):
+        # filter out 'None' values
+        filtered_access_control = list(
+            filter(
+                lambda access_control_entity: access_control_entity is not None,
+                access_control,
+            )
+        )
+
+        return {
+            "query": {
+                "template": {"params": {"access_control": filtered_access_control}},
+                "source": {
+                    "bool": {
+                        "filter": {
+                            "bool": {
+                                "should": [
+                                    {
+                                        "bool": {
+                                            "must_not": {
+                                                "exists": {"field": ACCESS_CONTROL}
+                                            }
+                                        }
+                                    },
+                                    {
+                                        "terms": {
+                                            f"{ACCESS_CONTROL}.enum": filtered_access_control
+                                        }
+                                    },
+                                ]
+                            }
+                        }
+                    }
+                },
+            }
+        }
+
+    def _is_active_atlassian_user(self, user_info):
+        user_url = user_info.get("self")
+        if not user_url:
+            self._logger.debug(
+                f"Skipping user: {user_info.get('displayName', 'user')} due to the absence of a personal URL."
+            )
+            return False
+
+        if not user_info.get("active"):
+            self._logger.debug(
+                f"Skipping user: {user_info.get('displayName', 'user')} as it is inactive or deleted."
+            )
+            return False
+
+        if user_info.get("accountType") != ATLASSIAN:
+            self._logger.debug(
+                f"Skipping user: {user_info.get('displayName', 'user')} because the account type is {user_info.get('accountType')}. Only 'atlassian' account type is supported."
+            )
+            return False
+
+        return True
+
+    def _decorate_with_access_control(self, document, access_control):
+        if self._dls_enabled():
+            document[ACCESS_CONTROL] = list(
+                set(document.get(ACCESS_CONTROL, []) + access_control)
+            )
+
+        return document
+
+    async def _user_information_list(self, project):
```

Comment:
suggestion: I would suggest that we move this method to the `JiraClient` class. I would also hide the low-level methods `api_call` and `paginated_api_call` methods then.

Reply:
I'll address this change in a separate PR. Created an issue regarding the same - https://github.com/elastic/connectors-python/issues/1417